pygmt/datasets/tile_map.py

```diff
@@ -147,6 +147,6 @@ def load_tile_map(region, zoom="auto", source=None, lonlat=True, wait=0, max_ret
 
     # If rioxarray is installed, set the coordinate reference system
     if hasattr(dataarray, "rio"):
-        dataarray = dataarray.rio.write_crs(input_crs="EPSG:3857")
+        dataarray = dataarray.rio.set_crs(input_crs="EPSG:3857")
```

Comment:
Changed from using [`.rio.write_crs`](https://corteva.github.io/rioxarray/stable/rioxarray.html#rioxarray.rioxarray.XRasterBase.write_crs) to [`.rio.set_crs`](https://corteva.github.io/rioxarray/stable/rioxarray.html#rioxarray.rioxarray.XRasterBase.set_crs) to prevent the CF coordinate `spatial_ref` from being written to the output `xarray.DataArray`. This fixes the int32 on Windows bu int64 on Linux/macOS issue mentioned in https://github.com/GenericMappingTools/pygmt/pull/2394#issuecomment-1484348012.

If this looks ok (and the tests pass), I'll proceed to merge in this PR.

Reply:
Looks good.